src/Features/Blockcore.Features.Wallet/Controllers/WalletRPCController.cs

```diff
@@ -46,6 +46,11 @@ public class WalletRPCController : FeatureController
         /// <summary>Wallet related configuration.</summary>
         private readonly WalletSettings walletSettings;
 
+        /// <summary>
+        /// The wallet name set by selectwallet method. This is static since the controller is a stateless type. This value should probably be cached by an injected service in the future.
+        /// </summary>
+        private static string CurrentWalletName;
```

Comment:
I think we should fix this now, it will be a fast fix maybe use as a hack put the field on the WallletManager just to avoid it being static?

Reply:
Ah I realize this is on RPC, it less crucial for me so this is fine (if it does not break tests)

Reply:
Doesn't break anything, and it is to give a custom support for multi-wallet in the RPC. It's a fairly important extra RPC method for anyone who relies on the RPC and have multiple wallets, but I don't think there are many that will ever use this. It is very useful when debugging and working with multiple wallets at the same time (normal wallet and special wallets).